src/session.rs

```diff
@@ -0,0 +1,342 @@
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::ffi::OsString;
+use std::fs;
+use std::io::{self, Write};
+use std::mem;
+use std::path::PathBuf;
+
+use anyhow::{anyhow, Context, Result};
+use reqwest::header::{HeaderMap, AUTHORIZATION};
+use reqwest::Url;
+use serde::{Deserialize, Serialize};
+
+use crate::utils::test_mode;
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+enum Meta {
+    Xh { about: String, xh: String },
+    Httpie { httpie: String },
+}
+
+impl Default for Meta {
+    fn default() -> Self {
+        Meta::Xh {
+            about: "xh session file".into(),
+            xh: xh_version(),
+        }
+    }
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+struct Auth {
+    #[serde(rename = "type")]
+    auth_type: Option<String>,
+    raw_auth: Option<String>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Cookie {
+    value: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    expires: Option<i64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    path: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    secure: Option<bool>,
+}
+
+#[derive(Debug, Default, Serialize, Deserialize)]
+struct Content {
+    #[serde(rename = "__meta__")]
+    meta: Meta,
+    #[serde(skip_serializing)]
+    auth: Option<Auth>,
```

Comment:
To prevent `auth` from being an option, `Content` could be changed to an enum. However, that seems a bit over-engineering so I will skip it. 
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6ae6c75db2f2eb73f262ac08f6c65d58